V1235813

medium

# D3Vault removeLiquidator and removeRouter function remove an address which does not even exist in List. And does not give back any emit for error

## Summary
D3Vault removeLiquidator and removeRouter function remove an address which does not even exist in List. And does not give back any emit for error

## Vulnerability Detail

On foundry 
removing address(123) on  d3Vault.removeLiquidator(address(123)); and d3Vault.removeRouter(address(123));
does not give back any emit for error. The address does not exist on Liquidator and Router



## Impact
It should emit an error that address does not exist in data list

## Code Snippet
https://gist.github.com/ranevikram12/3f6857dc795ad8d76acbe9243ae177b2#file-gistfile1-txt-L72

function testRemoveFunction() public {
     
        assertEq(d3Vault.allowedLiquidator(address(123)), false);
        vm.prank(vaultOwner);
        d3Vault.removeLiquidator(address(123));
       
        assertEq(d3Vault.allowedRouter(address(123)), false);
        vm.prank(vaultOwner);
        d3Vault.removeRouter(address(123));
        
    }

## Tool used
Foundry

Manual Review

## Recommendation
